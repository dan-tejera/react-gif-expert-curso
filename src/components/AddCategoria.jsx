import { useState } from "react"

export const AddCategoria = ({ onNuevaCategoria }) => {
  
  const [inputValor, setInputValor] = useState('');

  const cambiandoElInputValor = ( evento ) => {
    // console.log( evento.target.value );
    setInputValor( evento.target.value )
  }

  const onSumbit = ( evento ) => {
    evento.preventDefault(); // Evita que se recargue la pagina al enviar el formulario
    //console.log( inputValor );

    if( inputValor.trim().length <= 1 ) return; // Evita que se agregue una categoria vacia o con solo espacios en blanco

    //setCategorias( (categorias) => [inputValor, ...categorias] ); // ... es el operador spread, que permite copiar el contenido de un array en otro.

    onNuevaCategoria( inputValor ); // Llama a la funcion onNuevaCategoria que se pasa como prop desde el componente padre GifExpertApp

    setInputValor(''); // Limpia el input al enviar el formulario
  }

  return (
    <form onSubmit={ onSumbit }>
        <input
            type="text"
            placeholder="Buscar gifs"
            value={ inputValor }
            onChange={ cambiandoElInputValor }
        />
    </form>
  )
}
