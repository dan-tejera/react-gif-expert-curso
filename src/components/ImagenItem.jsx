import React from 'react'
import PropTypes from 'prop-types';

export const ImagenItem = ( {titulo, url, id} ) => {
  return (
    <div className='card'>
        <img src={ url } alt={ titulo }/>
        <p>{ titulo }</p>

    </div>
  )
}

ImagenItem.propTypes = {
  titulo: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
