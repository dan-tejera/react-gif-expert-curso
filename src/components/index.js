// Archivo de barril con las exportaciones de los componentes //
export * from './AddCategoria';
export * from './GifGrid';
export * from './ImagenItem';