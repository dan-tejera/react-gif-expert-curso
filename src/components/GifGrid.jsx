import { ImagenItem } from './ImagenItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ categoria }) => {  
  
  // Custom hook definido en el archivo useFetcGifs.js //
  const { imagenes, isLoading } = useFetchGifs( categoria );

  return (
    <>
        <h3>{ categoria }</h3>
        {
            isLoading && (<h2>Cargando...</h2>)
        }
        <div className='card-grid'>
          { 
            imagenes.map( ( imagen ) => (
              <ImagenItem 
                    key={ imagen.id }
                    { ...imagen } // para pasarle todas las propiedades del objeto imagen al componente ImagenItem
              />
            ))            
          }
        </div>
    </>
  )
}
