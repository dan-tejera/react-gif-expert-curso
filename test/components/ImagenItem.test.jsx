import { render, screen } from "@testing-library/react";
import { ImagenItem } from "../../src/components/ImagenItem";

const titulo = 'Titulo para pruebas';
const url = 'https://www.pruebas.com/imagenitem.html';

describe('Pruebas del componente <ImagenItem />', () => {

    test('Debe de hacer un match con el snapshot', () => {
        
        const { container } = render(<ImagenItem titulo={ titulo } url={ url } />);
        
        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el url y el alt indicado', () => {
         
        const { container } = render(<ImagenItem titulo={ titulo } url={ url } />);
        screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( titulo );
        // Cuando evalamos más de una propiedad del mismo elemento, es mejor desestructurarlo
        // Desesctructoruramos //
        const { src, alt } = screen.getByRole('img'); // getByRole buscar por atributo en el render del componente
        expect( src ).toBe( url );
        expect( alt ).toBe( titulo );
    });

    test('Debe de existir el título en el componente', () => {

        const { container } = render(<ImagenItem titulo={ titulo } url={ url } />);
        expect( screen.getByText( titulo ) ).toBeTruthy(); //getByText buscar por texto en el render del componente
    });
});