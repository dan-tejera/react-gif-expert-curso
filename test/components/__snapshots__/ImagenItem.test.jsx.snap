// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <ImagenItem /> Debe de hacer un match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo para pruebas"
      src="https://www.pruebas.com/imagenitem.html"
    />
    <p>
      Titulo para pruebas
    </p>
  </div>
</div>
`;
